pages/blog/_posts/2020-03-30-catch-imsi-from-vowifi-session.md

```diff
@@ -0,0 +1,157 @@
+layout: blogpost
+permalink: /blog/vowifi-imsi-leak/
+title: VoWifi leaking IMSI
+date: 2020-03-30
+tags: telco 4g epdg vowifi networking security
+author: <a href="https://www.linkedin.com/in/rskjetlein/">Roger Skjetlein</a>
+---
+
+4g offers more services than the earlier generation such as 3g and 2g. One of
+the service that really have gained traction later years is VoLTE, that is Voice
+over LTE and later that we will go more in dept regarding security is VoWifi,
+Voice over Wifi.
+
+VoWifi is beneficial in terms of using any Wifi connection offering public
+internet access thus extending and improving the coverage and connectivity.
+Think of it in similar ways of building our own eNodeB cellular network, but
+using commodity wifi components instead and avoiding the strict regulation and
+licensing spectrum.
+
+## What is IMSI
+*The international mobile subscriber identity (IMSI) is a number that uniquely
+identifies every user of a cellular network.[1] It is stored as a 64-bit field
+and is sent by the mobile device to the network. It is also used for acquiring
+other details of the mobile in the home location register (HLR) or as locally
+copied in the visitor location register. To prevent eavesdroppers from
+identifying and tracking the subscriber on the radio interface, the IMSI is sent
+as rarely as possible and a randomly-generated TMSI is sent instead.*
+
+* [IMSI article](https://en.wikipedia.org/wiki/International_mobile_subscriber_identity)
+
+### Security implications
+The IMSI is a secret identifier stored on the sim and can be exploited in many
+ways once known and bound to sim so changing UE will not help.
+
+Examples:
+* Locating user (UE)
+* Intercepting calls
+* Intercepting SMS (stealing two factor pin eg.)
+* ..and more
+
+## How VoWifi works
+When your phone is connected to datanetwork and with volte and vowifi enabled
+the device (UE) establish sip session directly to packetgateway via 4g, and via
+public internet to epdg (evolved packet data gateway) which is in essence a
+ipsec (ikev2) termination using SIM-AKA to authenticate UE. The ipsec comes into
+play since epdg is exposed publicly.
+
+I won't go further in depth for volte and vowifi since there are already
+excellent articles about the matter:
+* [Voice over WLAN](https://en.wikipedia.org/wiki/Voice_over_WLAN)
+* [Voice over LTE](https://en.wikipedia.org/wiki/Voice_over_LTE)
+
+![VoWifi topology](/img/blog/vowifi-imsi-leak/vowifi.jpg)
+
+## EPDG exposed on the public internet
+The Evolved Packet Data Gateway needs to be publicly available on the internet
+since UE needs to access it from an arbitrary no-trusted connection. The ipsec
+will secure and encrypt both the data and maintain the integrity of the
+connection throughout the session.
+
+The UE finds the epdg termination point by looking dns records partly following
+a convention decided by 3gpp and typically looks like this in DNS:
+
+```
+epdg.epc.mnc001.mcc242.pub.3gppnetwork.org. 3488 IN A 77.16.1.126
```

Comment:
Please remove mention of the network. Anonymize MNC and MCC by replacing it with 999.
Also change the IP address to some of ours or private.

Reply:
Fixed